opencog/atomutils/Unify.cc

```diff
@@ -348,14 +355,126 @@ bool Unify::is_bound_to_ancestor(const Variables& variables,
 
 Handle Unify::substitute(BindLinkPtr bl, const TypedSubstitution& ts)
 {
-	// Perform alpha-conversion, this will work over values that are
-	// non variables as well
-	//
 	// TODO: make sure that ts.second contains the declaration of all
 	// variables
-	Handle h = bl->alpha_conversion(strip_context(ts.first), ts.second);
+	return substitute(bl, strip_context(ts.first), ts.second);
+}
+
+Handle Unify::substitute(BindLinkPtr bl, const HandleMap& var2val,
+                         Handle vardecl)
```

Comment:
Perhaps you could call the method "rewrite()" or "reduce()" or "beta_reduce()" or something like that?